statistics/handle/cache/internal/lfu/lfu_cache_test.go

```diff
@@ -0,0 +1,161 @@
+// Copyright 2022 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package lfu
+
+import (
+	"sync"
+	"sync/atomic"
+	"testing"
+	"time"
+
+	"github.com/pingcap/tidb/statistics/handle/cache/internal/testutil"
+	"github.com/stretchr/testify/require"
+)
+
+var (
+	mockCMSMemoryUsage  = int64(4)
+	mockTopNMemoryUsage = int64(64)
+	mockHistMemoryUsage = int64(289)
+)
+
+func TestLFUPutGetDel(t *testing.T) {
+	capacity := int64(100)
+	lru, err := NewLFU(capacity)
+	require.NoError(t, err)
+	mockTable := testutil.NewMockStatisticsTable(1, 1, true, false, false)
+	mockTableID := int64(1)
+	lru.Put(mockTableID, mockTable, false)
+	require.Eventually(t, func() bool {
+		v, ok := lru.Get(mockTableID, false)
+		return ok && mockTable == v
+	}, 1*time.Second, 100*time.Millisecond)
+
+	lru.Del(mockTableID)
+	v, ok := lru.Get(mockTableID, false)
+	require.False(t, ok)
+	require.Nil(t, v)
+}
+
+func TestLFUFreshMemUsage(t *testing.T) {
+	lru, err := NewLFU(1000)
+	require.NoError(t, err)
+	t1 := testutil.NewMockStatisticsTable(1, 1, true, false, false)
+	t2 := testutil.NewMockStatisticsTable(2, 2, true, false, false)
+	t3 := testutil.NewMockStatisticsTable(3, 3, true, false, false)
+	lru.Put(int64(1), t1, false)
+	lru.Put(int64(2), t2, false)
+	lru.Put(int64(3), t3, false)
+	time.Sleep(1 * time.Second)
+	require.Equal(t, lru.Cost(), 6*mockCMSMemoryUsage+6*mockCMSMemoryUsage)
```

Comment:
Why not use `t1.MemoryUsage().TotalMemUsage + t2.MemoryUsage().TotalMemUsage + t3.MemoryUsage().TotalMemUsage`?